src/core/ipns/validator.js

```diff
@@ -33,10 +35,12 @@ const verify = (publicKey, entry, callback) => {
     }
 
     // Validate EOL
-    const validityDate = Date.parse(validity.toString())
+    if (validityType === IpnsEntry.validityType.EOL) {
+      const validityDate = Date.parse(validity.toString())
```

Comment:
So it doesn't get lost in the buried review comments, this needs to explicitly check the date format.

Reply:
Once `js-ipns` initial implementation PR is merged, I will create an issue in there for tackling this with priority. I have been thinking a lot earlier and reading some stuff about `bigint` for js and I think that I reached a way to add the nanoseconds precision here. I also need to find a way to explicitly check the date format according to the RFC 3339

Reply:
I wouldn't feel comfortable merging this until this is fixed. Unfortunately, due to some mistakes in how we make signatures in IPNS, we rely on *precisely* validating the date for security.

Reply:
@Stebalien just added it to [js-ipns PR](https://github.com/ipfs/js-ipns/pull/1#pullrequestreview-131898357)!